junebug/plugin.py

```diff
@@ -0,0 +1,23 @@
+class JunebugPlugin(object):
+    '''Base class for all Junebug plugins'''
+
+    def start_plugin(self, config):
+        '''Can be overridden with any required startup code for the plugin.
+        Can return a deferred.
+
+        config - Config that Junebug was started with.'''
```

Comment:
Should we maybe do this in the sphinx `:param:` format here so we can autogen? Not too important though.